docs/examples/model-energy.ipynb

```diff
@@ -0,0 +1,664 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Estimating model energy\n",
+    "\n",
+    "[![Open In\n",
+    "Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/nengo/keras-spiking/blob/master/docs/examples/model-energy.ipynb)\n",
+    "\n",
+    "One of the main motivations for using spiking methods is the potential for significant\n",
+    "energy savings over standard techniques. Thus it is useful to be able to estimate how\n",
+    "much energy would be used by a model on different devices, so that we can get an\n",
+    "idea of how different model/device parameters affect the energy usage before pursuing a\n",
+    "full deployment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "import nengo_dl\n",
+    "import nengo_loihi\n",
```

Comment:
I like the idea of showing deploying to Loihi, but I'm not keen on making this whole notebook dependent on NengoDL and NengoLoihi being installed. Maybe we could move those imports to the end (where they're actually used), so the first part of the notebook could run without them. I know that's non-traditional, but I think it makes sense in the context of a notebook where users could still see results from the first cells.

Reply:
I suspect that most people are just going to be looking at the rendered documentation, and if they are running the notebook themselves they can just `pip install` those two packages (if they've done `pip install keras-spiking[docs]` they have them already). Or if they really don't want to do that, they can just comment out those imports. But I also don't think it really matters if the imports are farther down, so I'm fine with moving them 🤷.

Reply:
You're right, I guess they're not that bad to pip install. I just ran into this because I had a nengo_loihi version installed that was incompatible with my Nengo version (which obviously isn't typical). But I don't see a reason to not have them later, either, so I think it makes sense for those odd cases that it's helpful to someone.